### PR TITLE
Ce 2980 TemplateClassification modal design

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/CE-2980

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js [hunk 1]
```diff
@@ -52,39 +52,30 @@ define('TemplateClassificationLabeling',
...
+			function chooseType() {
+				return mw.message('template-classification-edit-modal-title-select-type').escaped();
 			}
 
 			return {
-				addTemplate: addTemplate,
+				addTemplate: chooseType,
 				editType: editType,
-				addTypeBeforePublish: addTypeBeforePublish
+				addTypeBeforePublish: chooseType
```
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js:66]: I think that the name of the surfaced method should be changed respectively.

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js [hunk 2]
```diff
@@ -52,39 +52,30 @@ define('TemplateClassificationLabeling',
...
 			function editType(content) {
-				return content;
+				var helpText = $('<p></p>').addClass('instructions')
+					.html(mw.message('template-classification-edit-modal-help').parse())[0].outerHTML;
+				return helpText + content;
 			}
 
 			return {
-				addTemplate: addTemplate,
+				addTemplate: editType,
```
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js:78]: I think that the name of the surfaced method should be changed respectively.

File: extensions/wikia/TemplateClassification/templates/TemplateClassificationViewPageEntryPoint.mustache [hunk 3]
```diff
@@ -1,4 +1,3 @@
 {{templateTypeLabel}}
 <span class="template-classification-type-text" data-type="{{templateType}}">{{templateTypeName}}</span>
 {{#editButton}}<a class="template-classification-edit sprite-small edit"></a>{{/editButton}}
-
```
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/templates/TemplateClassificationViewPageEntryPoint.mustache:4]: Was it a spare new line? :)

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js [hunk 4]
```diff
@@ -52,39 +52,30 @@ define('TemplateClassificationLabeling',
...
-			function addTemplate(content) {
-				var $subtitle = $('<h2>').html(
-					mw.message('template-classification-edit-modal-select-type-sub-title').escaped()
-				);
-				return $subtitle[0].outerHTML + content;
-			}
-
 			function editType(content) {
-				return content;
+				var helpText = $('<p></p>').addClass('instructions')
```
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js]: Isn't it a bit too generic class name?

File: extensions/wikia/TemplateClassification/styles/TemplateClassification.scss [hunk 5]
```diff
@@ -19,9 +19,39 @@
 }
 
 .template-classification-edit-modal {
-	line-height: 2.5em;
+	/* Override modal defaults to new design */
+	font-size: 14px;
+	line-height: 20px;
+	color: #656e78;
```
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/styles/TemplateClassification.scss]: I'd consider defining the colors as variables.

**Comment by adamkarminski** [comment]
A few minor comments but LGTM in general. :)

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js (hunk 1 above)
Comment (kamilkoterba) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js:66]: Left part is for invocation and tells about the place where it's called. The right part tells about the way it's handled within Labeling extension. So it's adding choose type label to modal forced on publish.
Does that make sense now?

File: extensions/wikia/TemplateClassification/templates/TemplateClassificationViewPageEntryPoint.mustache (hunk 3 above)
Comment (kamilkoterba) [inline extensions/wikia/TemplateClassification/templates/TemplateClassificationViewPageEntryPoint.mustache:4]: yup.

File: extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js (hunk 1 above)
Comment (adamkarminski) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js:66]: I still think that the Labeling extension should be unaware of the places it is called at and provide a clear and self-explenatory interface. Imagine that we'll want to use the extension in 2 new places. Will you add 2 more methods to the interface that just execute the `chooseType` logic or wouldn't it be better for them to call the same method and make it consistent?

Comment (kamilkoterba) [inline extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js:66]: Maybe I misworded. Whole point of Labeling ext here is to provide modes that TC modal can pick. Please lookup var availableModes https://github.com/Wikia/app/blob/2d2ed7cd1628767bfac087845f99afd3be37a09f/extensions/wikia/TemplateClassification/scripts/TemplateClassificationLabeling.js
I consider this naming suits best. It explains what the mode is about so you can setup different labels for title button and content. Please note that combinations of labels are different for each mode so we can't merge any of them right now.
I'm also open for any naming improvements if you have a better idea or what do you think about moving on with this.

**Comment by kamilkoterba** [comment]
Tested.

**Comment by adamkarminski** [comment]
LGTM

**Pull request merged**